packages/sitecore-jss-nextjs/src/editing/editing-render-middleware.ts

```diff
@@ -96,9 +96,10 @@ export class ChromesHandler extends RenderMiddlewareBase {
     try {
       // Extract data from EE payload
       const editingData = this.extractEditingData(req);
-
+      // use https for requests with auth but also support unsecured http rendering hosts
+      const secure = req.headers.authorization || process.env.VERCEL ? true : false;
```

Comment:
why make this so complex, you could have just added an OR check in the `defaultResolveServerUrl`

```
private defaultResolveServerUrl = (req: NextApiRequest) => {
    return `${process.env.VERCEL || req.headers.authorization ? 'https' : 'http'}://${req.headers.host}`;
  };
```